docs/modules_overview.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: index
+title: SW-Helpers
+navigation_weight: 0
+---
+
+<!-- DO NOT EDIT. This page is autogenerated. --><!-- To make changes, edit templates/README.hbs, not README.md! -->
+[![Travis Build Status][travis-image]][travis-url]
+[![AppVeyor Build status][appveyor-image]][appveyor-url]
+[![Dependency Status][dependency-image]][dependency-url]
+[![Dev Dependency Status][dev-dependency-image]][dev-dependency-url]
+
+# SW-Helpers
+
+## Contents
+
+- [Ways to Use the Libraries](#ways-to-use-the-libraries)
+- [Setting Up](#setting-up)
+- [The Libraries](#the-libraries)
+  - [sw-appcache-behavior](#sw-appcache-behavior)
+  - [sw-background-sync-queue](#sw-background-sync-queue)
+  - [sw-broadcast-cache-update](#sw-broadcast-cache-update)
+  - [sw-build](#sw-build)
+  - [sw-cache-expiration](#sw-cache-expiration)
+  - [sw-cacheable-response](#sw-cacheable-response)
+  - [sw-cli](#sw-cli)
+  - [sw-lib](#sw-lib)
+  - [sw-offline-google-analytics](#sw-offline-google-analytics)
+  - [sw-precaching](#sw-precaching)
+  - [sw-routing](#sw-routing)
+  - [sw-runtime-caching](#sw-runtime-caching)
+- [External Contributions](#external-contributions)
+- [License](#license)
+
+## Ways to Use the Libraries
+
+Despite the number of modules in this package, there are a number of approaches to using these tools which many be used separately or together in any number of combinations.
+
+* **Precaching, runtime caching and routing**: Use [sw-lib](#sw-lib) to quickly
+  implement a [caching strategies](https://developers.google.com/web/fundamentals/instant-and-offline/offline-cookbook/) or a route.
+* **Build process**: Generate a service worker or manifest in your build process
+  using [sw-build](#sw-build).
+* **Command line**: Generate a service worker or manifest from the command line.
+* **Go beyond the basics**: Implement more advanced capabilities and more
+  sophisticated use cases using any of the other libraries, which are [described
+  below](#the-libraries).
+
+## Setting Up
+
+Each module is installed separately using the command line [as listed below](#the-libraries). To use a particular module:
+
+1. Install the module. For example:
+
+   `npm install --save-dev sw-lib`
+2. Copy the module's JavaScript and map files to your serving directory. To work properly these must be in the same directory as your service worker. For example:
```

Comment:
These don't *need* to be copied into the same directory as your top-level service worker. They can be located anywhere that's within the root of your web site; you just need to pass in the correct URL when calling `importScripts()`.

Cf. https://github.com/GoogleChrome/sw-helpers/blob/master/packages/sw-routing/test/sw/express-route.js#L5